packages/playground/data-liberation/src/entity-readers/WP_WXR_Sorted_Entity_Reader.php

```diff
@@ -0,0 +1,669 @@
+<?php
+
+use WordPress\ByteReader\WP_Byte_Reader;
+
+/**
+ * Data Liberation API: WP_WXR_Sorted_Entity_Reader class
+ *
+ * The topological sorted WXR reader class. This is an extension of the
+ * WP_WXR_Entity_Reader class that emits entities sorted topologically so that
+ * the parents are always emitted before the children.
+ *
+ * ## Implementation
+ *
+ * We create a custom table that contains the IDs and the new IDs created in the
+ * target system sorted in the parent-child order.
+ *
+ * This class extends the WP_WXR_Entity_Reader class and overrides the
+ * read_next_entity function to emit the entities in the correct order.
+ *
+ * List of entities      Sort order
+ * entity 1              entity 1          3
+ * entity 2, parent 1    └─ entity 2       2
+ * entity 3, parent 2       └─ entity 3    1
+ * entity 4, parent 2       └─ entity 4    1
+ *
+ * List of entities      Sort order
+ * entity 4, parent 2    entity 1          3
+ * entity 3, parent 2    └─ entity 2       2
+ * entity 2, parent 1       └─ entity 3    1
+ * entity 1                 └─ entity 4    1
+ *
+ * List of entities      Sort order
+ * entity 1              entity 1          3
+ * entity 3, parent 2    └─ entity 2       2
+ * entity 2, parent 1       └─ entity 3    1
+ *
+ * List of entities      Sort order
+ * entity 1              entity 1          1
+ * entity 2              entity 2          1
+ * entity 3              entity 3          1
+ *
+ * @since WP_VERSION
+ */
+class WP_WXR_Sorted_Entity_Reader extends WP_WXR_Entity_Reader {
+
+	/**
+	 * The base name of the table used to store the IDs, the new IDs and the
+	 * sort order.
+	 */
+	const TABLE_NAME = 'data_liberation_map';
+
+	/**
+	 * The current database version, to be used with dbDelta.
+	 */
+	const DB_VERSION = 1;
+
+	/**
+	 * The current session ID.
+	 */
+	protected $current_session = null;
```

Comment:
Nit: Let's clarify the word "session" as it can have multiple meanings in a PHP app

```suggestion
	 * The current import session ID.
	 */
	protected $import_session_id = null;
```